modyn/trainer_server/internal/trainer/pytorch_trainer.py

```diff
@@ -294,15 +295,16 @@ def train(self) -> None:  # pylint: disable=too-many-locals, too-many-branches
                 pre_downsampling_size = target.shape[0]
 
                 with torch.autocast(self._device_type, enabled=self._amp):
-                    if self._downsampling_enabled:
+                    if self._downsampling_enabled and not self._sample_before_batch:
                         # TODO(#218) Persist information on the sample IDs/weights when downsampling is performed
                         assert self._downsampler is not None
                         big_batch_output = self._model.model(data)
-                        downsampled_indexes, weights = self._downsampler.sample(big_batch_output, target)
+                        downsampled_indexes, weights = self._downsampler.batch_then_sample(big_batch_output, target)
```

Comment:
Ok I don't understand why we need a separate method call here. IIRC, we discussed that a downsampler takes as input samples and outputs fewer samples. Batch then sample just calls this function every batch, sample then batch calls it before training. So we use the same downsampler interface. But now we do not have the same interface anymore. Is there a reason?